cubeviz/tools/collapse_cube.py

```diff
@@ -0,0 +1,359 @@
+        vbl.addLayout(hb_region)
+        vbl.addLayout(hb_start)
+        vbl.addLayout(hb_end)
+        vbl.addLayout(hbl_error)
+        vbl.addLayout(hb_buttons)
+
+        # Fire the callback to set the default values for everything
+        self._region_selection_change(0)
+
+        self.setLayout(vbl)
+        self.setMaximumWidth(700)
+        self.show()
+
+    def _region_selection_change(self, index):
+        """
+        Callback for a change on the region selection combo box.
+
+        :param newvalue:
+        :return:
+        """
+
+        newvalue = self.region_combobox.currentText()
+
+        # First, let's see if this is one of the custom options
+        if 'Custom' in newvalue and 'Wavelength' in newvalue:
+            # Custom Wavelengths
+            self.hide_start_end(False)
+            self.start_label.setText("Start Wavelength:")
+            self.end_label.setText("End Wavelength:")
+
+        elif 'Custom' in newvalue and 'Indices' in newvalue:
+            # Custom indices
+            self.hide_start_end(False)
+            self.start_label.setText("Start Index:")
+            self.end_label.setText("End Index:")
+
+        else:
+            # Region defined in specviz
+            self.hide_start_end(True)
+
+            # We are going to store the start and end wavelengths in the text boxes even though
+            # they are hidden. This way we can use the text boxes later as a hidden storage container.
+            # TODO: Should probably save the ROIs so the start and end values are more accurate.
+            regex = r"-?(?:0|[1-9]\d*)(?:\.\d*)?(?:[eE][+\-]?\d+)?"
+            floating = re.findall(regex, newvalue)
+            self.start_text.setText(floating[0])
+            self.end_text.setText(floating[1])
+
+        # Let's update the text on the widget
+        if 'Custom' in newvalue:
+            self.widget_desc.setText(self._general_description + "\n\n" + self._custom_description)
+        else:
+            self.widget_desc.setText(self._general_description)
+
+    def hide_start_end(self, dohide):
+        """
+        Show or hide the start and end indices depending if the region
+        is defined from the specviz plot OR if we are using custom limits.
+
+        :param dohide:
+        :return:
+        """
+        if dohide:
+            self.start_label.hide()
+            self.start_text.hide()
+            self.end_label.hide()
+            self.end_text.hide()
+        else:
+            self.start_label.show()
+            self.start_text.show()
+            self.end_label.show()
+            self.end_text.show()
+
+    def calculate_callback(self):
+        """
+        Callback for when they hit calculate
+        :return:
+        """
+
+        # Grab the values of interest
+        data_name = self.data_combobox.currentText()
+        start_value = float(self.start_text.text().strip())
+        end_value = float(self.end_text.text().strip())
```

Comment:
There needs to be some error checking on the input strings here. Right now if these fields are empty, or if the values are invalid, we just get a `ValueError: could not convert string to float`. Maybe it would be useful to have the same kind of behavior that we implemented for the slice and wavelength value fields where invalid values turn red and are not accepted (actually it looks like this is done below, but the error handling logic needs to happen earlier).

Reply:
will be fixed in the next commit